velox/dwio/common/compression/Compression.h

```diff
@@ -66,14 +66,20 @@ class Decompressor {
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param isRawUncompress specify whether to perform a raw uncompress
+ * @param compressedLength the compressed block length for raw uncompress
+ * @param zlibWindowBits window bits to use for zlib decompression
  */
 std::unique_ptr<dwio::common::SeekableInputStream> createDecompressor(
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool isRawUncompress = false,
+    size_t compressedLength = 0,
+    int zlibWindowBits = -15);
```

Comment:
Can we create a struct to encapsulate the format specific options instead of passing them directly in the generic interface?  Passing something like this should be ok for now:

```c++
struct FormatOptions {
  union {
    int zlibWindowBits = -15;
  } zlib;
};
```

Reply:
@Yuhta Updated the PR. Do you have any other comments?